test/inductor/test_triton_kernels.py

```diff
@@ -1677,6 +1677,47 @@ def fn(x):
             self.assertEqual(out_e[0], out_c[0])
             self.assertEqual(out_e[1], out_c[1])
 
+    # TODO enable this test case on XPU.
+    @requires_cuda
+    def test_i64_input(self):
+        # The i64 "seed" input needs to be marked as "i64", not "i32".
+        @triton.jit
+        def triton_add_noise_(x_ptr, y_ptr, seed, numel, BLOCK_SIZE: tl.constexpr):
+            pid = tl.program_id(0)
+            offsets = pid * BLOCK_SIZE + tl.arange(0, BLOCK_SIZE)
+
+            x = tl.load(x_ptr + offsets, mask=(offsets < numel))
+            rnd = tl.rand(seed, offsets)
+            res = x + rnd
+            tl.store(y_ptr + offsets, res, mask=(offsets < numel))
+
+        def add_noise(x, seed):
+            y = torch.empty_like(x)
+            numel = x.numel()
+            BLOCK_SIZE = 256
+
+            def grid(meta):
+                return (triton.cdiv(numel, meta["BLOCK_SIZE"]),)
+
+            triton_add_noise_[grid](x, y, seed, numel, BLOCK_SIZE)
+            return y
+
+        def fn(x):
+            x = x * x
+            seed = torch.randint(
+                low=0, high=2**62, size=(1,), dtype=torch.int64
+            ).item()
```

Comment:
Do we want to explicitly put this in the above int32 range, so that we don't get errors passing silently? Or maybe check both within and outside int32?

Reply:
Good point, updated it to `[2**32, 2**62)`.

Reply:
Thanks for doing this, although I realized how insanely unlikely it is sample int32 from [2, 2**62] :)